### DISCUSSION TITLE
Lossy conversion of E2E files

### DISCUSSION THREAD

**Discussion by ghost** [marksgraham/OCT-Converter#64] [General]
Hello, 
I'm having some trouble with opening the E2E files that I have. 
It does create tiff or jpg files as I want, but the images have worse resolution than when it was imaged. 
I'm guessing it's because of the error message "Lossy conversion from float64 to uint8.". But I'm not sure how I can convert the e2e to uint8 prior to getting the images.  
Could you give me some advice on that?? 
Thank you for your help!! 
![python](https://user-images.githubusercontent.com/80532121/184939729-f9b05972-8657-4203-8b9d-5d64d83fc615.PNG)

**Comment by marksgraham** [comment]
Hi,

You can get the raw data as a list of numpy arrays doing something like

```python
file = E2E(filepaths[0]['path'])
oct_volumes = (
    file.read_oct_volume()
)  # returns a list of all OCT volumes with additional metadata if available
for volume in oct_volumes:
    list_of_slices = volume.volume
    slice_as_np_array = list_of_slices[0]
```

and from there apply any conversion you want / save or plot the images as you like.
I don't think the conversion to uint8 would be causing resolution issues though. What exactly do you mean - the images look less sharp?

**Reply by ghost** [reply]
Hi, thank you for the help. 
When I use the function, the images look darker than it does on the software. 
And this is probably because I'm changing the contrast, but it does look less smooth and more grainy. 
I'm guessing this might be because of those loss errors. Is there a reason why when I covert the images to tiff, it wouldn't accept float64 type data??

**Reply by marksgraham** [reply]
I think the reason the images are looking darker is due to the issue discussed here:
https://github.com/marksgraham/OCT-Converter/issues/21
So it may be worth experimenting with the gamma value
